storage/innobase/xtrabackup/src/redo_log.cc

```diff
@@ -620,8 +620,7 @@ void Archived_Redo_Log_Monitor::skip_for_block(lsn_t lsn, uint32_t no,
     for (auto ptr = reader.get_buffer(); ptr < reader.get_buffer() + len;
          ptr += OS_FILE_LOG_BLOCK_SIZE, bytes_read += OS_FILE_LOG_BLOCK_SIZE) {
       auto arch_block_no = log_block_get_hdr_no(ptr);
-      auto arch_block_checksum = log_block_get_checksum(ptr);
-      if (no == arch_block_no && checksum == arch_block_checksum) {
+      if (no == arch_block_no) {
         msg("xtrabackup: Archived redo log has caught up\n");
```

Comment:
you should calculate checksum from the archive block data and also from the header. It should match. If it doesn't, thenkeep retrying until checksum matches

Reply:
We already have a check to error out if mismatch in checksum.
https://github.com/percona/percona-xtrabackup/blob/ec3a018bbb85407e026561cbd3705b80e0a42a6e/storage/innobase/xtrabackup/src/redo_log.cc#L533

Reply:
I think you should skip checksum verification only len mismatch

And in the case of length mismatch, you could skip comparison. But there should be retry logic. Release assert(crash) is not a good idea.

Reply:
length mismatch is done.